tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,877 @@
+                "providerState", providerStateAfterSending
+            )
+            Ok(finalState)
+        }
+    }
+
+    pure def endAndBeginBlockForConsumer(
+        currentState: ProtocolState,
+        chain: Chain,
+        // by how much the timestamp of the chain should be advanced for the next block
+        timeAdvancement: Time): Result = {
+            if (not(currentState.consumerStates.keys().contains(chain))) {
+                Err("chain is not a consumer")
+            } else {
+                // if the chain is not a consumer, return an error
+                val currentConsumerState = currentState.consumerStates.get(chain)
+                val newChainState = currentConsumerState.chainState.endAndBeginBlockShared(timeAdvancement)
+                val newConsumerState = currentConsumerState.with(
+                    "chainState", newChainState
+                )
+                val maturedPackets = newConsumerState.maturationTimes.keys().filter(
+                    packet => 
+                    val maturationTime = newConsumerState.maturationTimes.get(packet)
+                    maturationTime <= newChainState.lastTimestamp
+                )
+                val newMaturationTimes = newConsumerState.maturationTimes.mapRemoveAll(maturedPackets)
+                val newOutstandingPackets = newConsumerState.outstandingPacketsToProvider.concat(
+                    maturedPackets.map(
+                        packet => {
+                            id: packet.id, 
+                            sendingTime: newConsumerState.chainState.lastTimestamp
+                        }
+                    ).toList()
+                )
+                val newConsumerState2 = newConsumerState.with(
+                    "maturationTimes", newMaturationTimes
+                ).with(
+                    "outstandingPacketsToProvider", newOutstandingPackets
+                )
+                val newConsumerStates = currentState.consumerStates.set(chain, newConsumerState2)
+                val newState = currentState.with(
+                    "consumerStates", newConsumerStates
+                )
+                Ok(newState)
+            }
+    }
+
+    // ===================
+    // UTILITY FUNCTIONS
+    // which do not hold the core logic of the protocol, but are still part of it
+    // ===================
+
+    // Get the running consumer that is furthest away from the provider in terms of timestamp.
+    pure def getFurthestConsumer(currentState: ProtocolState): Chain = {
+        val furthestConsumer = getRunningConsumers(currentState.providerState).MaxBy(
+            consumer => abs(
+                currentState.providerState.chainState.lastTimestamp -
+                currentState.consumerStates.get(consumer).chainState.lastTimestamp
+            )
+        )
+        furthestConsumer
+    }
+
+    pure def getRunningConsumersFromMap(consumerStatus: Chain -> str): Set[Chain] = {
```

Comment:
Maybe I missed this, but is this `def` and `getStoppedConsumersFromMap` used somewhere?

Reply:
It's not, you are right :) removed!